src/MuxManager.cpp

```diff
@@ -234,6 +234,33 @@ void MuxManager::processProbeMuxState(const std::string &portName, const std::st
     }
 }
 
+//
+// ---> addOrUpdateDefaultRouteState(boost::asio::ip::address& address, const std::string &routeState);
+//
+// update default route state based on state db notification
+//
+void MuxManager::addOrUpdateDefaultRouteState(boost::asio::ip::address address, const std::string &routeState) 
+{
+    MUXLOGINFO(boost::format("%s: default route state: %s") % address % routeState);
+
+    if (address.is_v4()) {
+        mIpv4DefaultRouteState = routeState;
+    } else {
+        mIpv6DefaultRouteState = routeState;
+    }
+
+    std::string nextState = "na";
+    if (mIpv4DefaultRouteState == "ok" && mIpv6DefaultRouteState == "ok") {
```

Comment:
@lguohan do we have concern if either IPv4 or IPv6 BGP is not configured? Can we assume we will have them both eventually?

Reply:
Updated per discussion in Gemini sync-up meeting.